crates/bevy_render/src/lib.rs

```diff
@@ -120,28 +120,21 @@ impl Plugin for RenderPlugin {
 
         if let Some(backends) = options.backends {
             let instance = wgpu::Instance::new(backends);
-            let surface = {
-                let world = app.world.cell();
-                let windows = world.get_resource_mut::<bevy_window::Windows>().unwrap();
-                let raw_handle = windows.get_primary().map(|window| unsafe {
-                    let handle = window.raw_window_handle().get_handle();
-                    instance.create_surface(&handle)
-                });
-                raw_handle
-            };
             let request_adapter_options = wgpu::RequestAdapterOptions {
                 power_preference: options.power_preference,
-                compatible_surface: surface.as_ref(),
+                compatible_surface: None,
```

Comment:
Setting `compatible_surface` is necessary for webgl2 support. You can only create a webgl2 context by calling `.getContext("webgl2")` on a canvas element. The resulting context is then limited to operate on this canvas, so creating a dummy canvas to call `.getContext("webgl2")` doesn't work. See https://github.com/bevyengine/bevy/pull/3039.